os/net/netmgr/netdev_wifi.c

```diff
@@ -99,3 +99,30 @@ int netdev_handle_wifi(struct netdev *dev, lwnl_req cmd, void *data, uint32_t da
 	}
 	return res;
 }
+
+int trwifi_serialize_scaninfo(uint8_t **buffer, trwifi_scan_list_s *scan_list)
+{
+	trwifi_scan_list_s *item = scan_list;
+	int cnt = 0, total = 0;
+	while (item) {
+		item = item->next;
+		cnt++;
+	}
+	total = cnt;
```

Comment:
I think you can put `sizeof(trwifi_ap_scan_info_s) * cnt` to `total` here.
A `total` is used for total size as calculation with `sizeof(trwifi_ap_scan_info_s)`, not used alone anymore.
These changes can get good performance by removing unnecessary repetition of function calls.
You can change these like below:
```
total_size = sizeof(trwifi_ap_scan_info_s) * total;
*buffer = (uint8_t *)kmm_malloc(total_size);
...
return total_size;
```